src/test/test-ns.c

```diff
@@ -47,6 +47,8 @@ int main(int argc, char *argv[]) {
         };
 
         static const NameSpaceInfo ns_info = {
+                .tmp = (char *) "/tmp/systemd-private-XXXXXX",
+                .var = (char *) "/var/tmp/systemd-private-XXXXXX",
```

Comment:
Nope, you cannot do that. You can keep the `tmp_dir` and `var_tmp_dir` stack variables and make `ns_info` non-static, and assign them here.